apps/voting/app/src/components/LocalIdentityBadge/LocalIdentityBadge.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { Badge, IdentityBadge, font } from '@aragon/ui'
+import { LocalIdentityModalContext } from '../LocalIdentityModal/LocalIdentityModalManager'
+import { IdentityContext } from '../IdentityManager/IdentityManager'
+
+const LocalIdentityBadge = ({ address, ...props }) => {
+  const { resolve } = React.useContext(IdentityContext)
+  const { showLocalIdentityModal, updates$ } = React.useContext(
+    LocalIdentityModalContext
+  )
+  const [label, setLabel] = React.useState()
+  const handleResolve = async () => {
+    try {
+      const { name = null } = await resolve(address)
+      setLabel(name)
+    } catch (e) {
+      // address does not ressolve to identity
+    }
+  }
+  const handleClick = () => {
+    showLocalIdentityModal(address)
+      .then(handleResolve)
+      .catch(e => {
+        /* user cancelled modify intent */
+      })
+  }
+  React.useEffect(() => {
+    handleResolve()
+    const subscription = updates$.subscribe(updatedAddress => {
+      if (updatedAddress.toLowerCase() === address.toLowerCase()) {
+        handleResolve()
+      }
+    })
+    return () => subscription.unsubscribe()
+  }, [])
```

Comment:
Same than https://github.com/aragon/aragon-apps/pull/722#discussion_r268723603:

```suggestion
  }, [name, address])
```